### DISCUSSION TITLE
Enable/Disable Hibernate based on custom configuration

### DISCUSSION THREAD

**Discussion by Dutch-0** [quarkusio/quarkus#31616] [Q&A] [answered]
How could I disable or enable Hibernate based on configuration property? 
For example, only enable hibernate if `tld.domain.myloggerapp.dest` is set to `=database`.

[a comment by an automated account is not shown]

**Comment by Sanne** [comment]
Would this address your needs?

https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.enabled

**Comment by Dutch-0** [comment]
> If Hibernate ORM is disabled during the build, all processing related to Hibernate ORM will be skipped, but it will not be possible to activate Hibernate ORM at runtime: quarkus.hibernate-orm.active will default to false and setting it to true will lead to an error.

Unfortunately that would not help

EDIT:
But maybe [quarkus.hibernate-orm.active](https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.active), I will try it out

**Comment by yrodiere** [comment, marked as answer]
You can enable/disable Hibernate ORM:

* at build time with [quarkus.hibernate-orm.enabled](https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.enabled) (if you disable Hibernate ORM that way, it cannot be made available at runtime again)
* on startup with [quarkus.hibernate-orm.active](https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.active)

I don't think there's a way to automatically translate`tld.domain.myloggerapp.dest=database` to the above, though. @radcortez might know.

**Reply by radcortez** [reply]
A way would be to provide a `ConfigSourceFactory`:
https://smallrye.io/smallrye-config/3.1.3/config-sources/custom/#configsourcefactory

You could read `tld.domain.myloggerapp.dest`, and match the value. Then you can return a `quarkus.hibernate-orm.active` to either `true` or `false` in a new source.

**Reply by Dutch-0** [reply]
[quarkus.hibernate-orm.active](https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.active) works perfectly. 

@radcortez ConfigSourceFactory Looks promising, will try to implement that also. Thank you
